dessia_common/datatools/modeling.py

```diff
@@ -0,0 +1,734 @@
+        return model.predict(inputs).tolist()
+
+    @classmethod
+    def fit_predict_(cls, inputs: List[List[float]], outputs: List[List[float]], predicted_inputs: List[List[float]],
+                    name: str = '', **hyperparameters):
+        model = cls.fit_(inputs, outputs, name, **hyperparameters)
+        return model, model.predict(predicted_inputs)
+
+    def score(self, inputs: List[List[float]], outputs: List[List[float]]):
+        model = self._instantiate_skl()
+        return model.score(inputs, outputs)
+
+
+class LinearRegression(BaseModel):
+    _standalone_in_db = True
+
+    def __init__(self, coef_: List[List[float]] = None, intercept_: List[List[float]] = None, name: str = ''):
+        self.coef_ = coef_
+        self.intercept_ = intercept_
+        BaseModel.__init__(self, name=name)
+
+    @classmethod
+    def _skl_class(cls):
+        return linear_model.Ridge
+
+    def _instantiate_skl(self):
+        model = self._call_skl_model()
+        model.coef_ = npy.array(self.coef_)
+        model.intercept_  = npy.array(self.intercept_)
+        return model
+
+    @classmethod
+    def _instantiate_dessia(cls, model, name: str = ''):
+        return cls(coef_=model.coef_.tolist(), intercept_=model.intercept_.tolist(), name=name)
+
+    @classmethod
+    def fit(cls, inputs: List[List[float]], outputs: List[List[float]], name: str = '',
+            alpha: float = 1., fit_intercept: bool = True, tol: float = 0.001):
+        return cls.fit_(inputs, outputs, name=name, alpha=alpha, fit_intercept=fit_intercept, tol=tol)
+
+    @classmethod
+    def fit_predict(cls, inputs: List[List[float]], outputs: List[List[float]], predicted_inputs: List[List[float]],
+                    name: str = '', alpha: float = 1., fit_intercept: bool = True, tol: float = 0.001):
+        return cls.fit_predict_(inputs, outputs, predicted_inputs, name=name,
+                                alpha=alpha, fit_intercept=fit_intercept, tol=tol)
+
+
+class BaseTree(BaseModel):
+
+    def __init__(self, n_classes: List[int] = None, n_features: int = None, n_outputs: int = None,
+                 tree_state: Dict[str, Any] = None, name: str = ''):
+        self.n_classes = n_classes
+        self.n_features = n_features
+        self.n_outputs = n_outputs
+        self.tree_state = tree_state
+        BaseModel.__init__(self, name=name)
+
+    def _data_hash(self):
+        hash_ = npy.linalg.norm(self.tree_state['values'][0])
+        hash_ += sum(self.n_classes)
+        hash_ += self.n_features
+        hash_ += self.n_outputs
+        return int(hash_ % 1e5)
+
+    @classmethod
+    def _skl_class(cls):
+        return tree._tree.Tree
+
+    def _call_skl_model(self):
+        return self._skl_class()(self.n_features, npy.array(self.n_classes), self.n_outputs)
+
+    @staticmethod
+    def _getstate_dessia(model):
+        state = model.__getstate__()
+        dessia_state = {'max_depth': int(state['max_depth'])}
```

Comment:
This is true for all dict. Should i do for every other dict (kwargs ones) in this module ?

About the State, i'm not sure it is useful since it is a requirement from sklearn CTree (the C version of the DecisionTree) and has just been a painful operation and moment to make it work with our constraints...

Reply:
I've changed all dicts and i keep state as it is now.